src/commands/environments/get.ts

```diff
@@ -25,24 +25,30 @@ export default class DetailedEnvironments extends Base {
         const { headless, project } = flags
         await this.requireProject(project, headless)
 
-        if (flags.headless && !keys) {
-            throw (new Error('In headless mode, the keys flag is required'))
-        }
+        let environments = await fetchEnvironments(this.authToken, this.projectKey)
         if (keys.length) {
-            let environments = await fetchEnvironments(this.authToken, this.projectKey)
             environments = environments.filter((environment) => 
                 keys.includes(environment.key) || keys.includes(environment._id))
             this.writer.showResults(environments)
-        } else {
-            const responses = await inquirer.prompt<EnvironmentPromptResult>(
-                [environmentPrompt],
-                {
-                    token: this.authToken,
-                    projectKey: this.projectKey
-                }
-            )
-            const environment = await fetchEnvironmentByKey(this.authToken, this.projectKey, responses.environment._id)
-            this.writer.showResults(environment)
+            return
+        } 
+
+        // show all environments if no keys flag provided in headless mode
+        if (flags.headless) {
+            this.writer.showResults(environments)
+            return
         }
+
+        // prompt for keys in interactive mode 
+        const responses = await inquirer.prompt<EnvironmentPromptResult>(
+            [environmentPrompt],
+            {
+                token: this.authToken,
+                projectKey: this.projectKey
+            }
+        )
+        const environment = await fetchEnvironmentByKey(this.authToken, this.projectKey, responses.environment._id)
```

Comment:
if you're fetching all of the environments anyway, does it make sense to just search those for this `_id` and save an api request?

Reply:
good point